src/Admin/AbstractAdmin.php

```diff
@@ -559,16 +529,9 @@ public function __construct($code, $class, $baseControllerName)
         $this->datagridValues['_per_page'] = $this->maxPerPage;
     }
 
-    /**
-     * {@inheritdoc}
-     *
-     * NEXT_MAJOR: return null to indicate no override
-     */
-    public function getExportFormats()
+    public function getExportFormats(): array
     {
-        return [
-            'json', 'xml', 'csv', 'xls',
-        ];
+        return [];
```

Comment:
This is right? NEXT_MAJOR is talking about `null` as default value

Reply:
I don't understand the comment `return null to indicate no override`.

But this was related to this
![image](https://user-images.githubusercontent.com/9052536/77827640-28bec980-7117-11ea-8fb1-d06dc10969df.png)

I agree that always returning an array is better.
The check should just be updated in `getAvailableFormats`, and it seems like it was.

Reply:
Yes, but I took a look at the

Reply:
I think we should allow `null` as a value. How could you disable the export completly?

Reply:
Do you mean if you don't want to see the export button? in[ base_list.html.twig there is this check](https://github.com/sonata-project/SonataAdminBundle/blob/3.x/src/Resources/views/CRUD/base_list.html.twig#L187):
```
{% if admin.hasRoute('export') and admin.hasAccess('export') and export_formats|length %}
```
I agree that `null` maybe express better this intention and `null|length` works, but for consistency I think it's better use `array` and since it's disabled by default (with the empty array) I don't think it's a problem. Or are you thinking something else?

Reply:
Yeah, I think we can disable export with an empty array.

Reply:
The override I was talking about in f31dea1a13aa5fe97edd2ade6321c1ae46f98916 is the override of formats returned by this method: https://github.com/sonata-project/exporter/blob/ed2b6e43982eec5c267b3ff3bc0e3c97ab98b849/src/Exporter.php#L75-L78

I was meaning to return an empty array for "no export formats", and `null` for "use whatever `Exporter::getAvailableFormats` returns, sorry if it was not clear.